common/hyperscaler/rules/simple_parser.go

```diff
@@ -0,0 +1,93 @@
+package rules
+
+import (
+	"fmt"
+	"strings"
+)
+
+type SimpleParser struct {
+}
+
+func (g *SimpleParser) Parse(ruleEntry string) (*Rule, error) {
+	outputRule := NewRule()
+
+	ruleEntry = strings.ReplaceAll(ruleEntry, " ", "")
+	ruleEntry = strings.ReplaceAll(ruleEntry, "\t", "")
+	ruleEntry = strings.ReplaceAll(ruleEntry, "\n", "")
+	ruleEntry = strings.ReplaceAll(ruleEntry, "\r", "")
+	ruleEntry = strings.ReplaceAll(ruleEntry, "\f", "")
+
+	outputInputPart := strings.Split(ruleEntry, "->")
+
+	if len(outputInputPart) > 2 {
+		return nil, fmt.Errorf("rule has more than one arrows")
+	}
+
+	inputPart := outputInputPart[0]
+
+	planAndInputAttr := strings.Split(inputPart, "(")
+
+	if len(planAndInputAttr) > 2 {
+		return nil, fmt.Errorf("rule has more than one '('")
+	}
+
+	forValidationOnly := strings.Split(inputPart, ")")
+
+	if len(forValidationOnly) > 2 {
+		return nil, fmt.Errorf("rule has more than one ')'")
+	}
+
+	if strings.Contains(inputPart, "(") && !strings.Contains(inputPart, ")") {
+		return nil, fmt.Errorf("rule has unclosed parentheses")
+	}
+
+	if !strings.Contains(inputPart, "(") && strings.Contains(inputPart, ")") {
+		return nil, fmt.Errorf("rule has unclosed parentheses")
+	}
+
+	_, err := outputRule.SetPlan(planAndInputAttr[0])
+	if err != nil {
+		return nil, err
+	}
+
+	if len(planAndInputAttr) > 1 {
+		inputPart := strings.TrimSuffix(planAndInputAttr[1], ")")
+
+		inputAttrs := strings.Split(inputPart, ",")
```

Comment:
If we use this code, we could extract some important characters/runes as constants with clear name.

Reply:
An example, please.

Reply:
```go
const COMMA_TOKEN = ","
const COMMA_RUNE = ","
```